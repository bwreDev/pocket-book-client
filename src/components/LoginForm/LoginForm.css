.LoginForm input {
  height: 40px;
  border-radius: 15px;
  border-style: none;
  margin: 5px;
  background-color: lightgreen;
}

.LoginForm fieldset {
  border-style: none;
  height: 250px;
}

.login-button {
  background-color: #225a2a;
  color: white;
  border-radius: 15px;
  padding: 20px;
  border: none;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 10px;
  width: 250px;
}

.login-button:hover {
  background-color: lightgreen;
  color: #303030;
  cursor: pointer;
}

.red {
  color: red;
}

@media only screen and (min-width: 600px) {
  .LoginForm input {
    margin-top: 20px;
    margin-bottom: 20px;
    height: 60px;
    font-size: 20px;
  }

  .LoginForm fieldset {
    height: 400px;
  }
}

@media only screen and (min-width: 1200px) {
  .LoginForm input {
    margin-top: 40px;
    margin-bottom: 40px;
    height: 80px;
    font-size: 24px;
  }

  .LoginForm fieldset {
    height: 600px;
  }
}
