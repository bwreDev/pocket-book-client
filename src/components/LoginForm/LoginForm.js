import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
import Navbar from '../../components/Navbar/Navbar';
import './LoginForm.css';

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {},
  };

  state = { error: null };

  handleSubmitJwtAuth = (input) => {
    input.prinputDefault();
    this.setState({ error: null });
    const { username, password } = input.target;

    AuthApiService.postLogin({
      username: username.value,
      password: password.value,
    })
      .then((res) => {
        username.value = '';
        password.value = '';
        this.props.onLoginSuccess();
        this.props.history.push('/user');
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;

    return (
      <>
        <Navbar />
        <form
          className='LoginForm'
          onSubmit={this.handleSubmitJwtAuth}
        >
          <div role='alert'>
            {error && <p className='red'>{error}</p>}
          </div>
          <fieldset>
            <h3>Login</h3>
            <label htmlFor='LoginForm__username'></label>
            <input
              name='username'
              type='text'
              placeholder='Username'
              id='LoginForm__username'
              required
            />
            <label htmlFor='LoginForm__password'></label>
            <input
              name='password'
              type='password'
              placeholder='Password'
              id='LoginForm__password'
            />
          </fieldset>
          <button className='login-button' type='submit'>
            Login
          </button>
        </form>
      </>
    );
  }
}
