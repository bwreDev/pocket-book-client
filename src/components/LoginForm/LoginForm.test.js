import React from 'react';
import ReactDOM from 'react-dom';
import LoginForm from './LoginForm';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <LoginForm />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
