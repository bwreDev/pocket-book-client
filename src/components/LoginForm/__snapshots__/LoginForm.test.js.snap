// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <nav
    className="navbar"
    role="navigation"
  >
    <ul
      className="navbar-list"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>,
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <fieldset>
      <h3>
        Login
      </h3>
      <label
        htmlFor="LoginForm__username"
      />
      <input
        id="LoginForm__username"
        name="username"
        placeholder="Username"
        required={true}
        type="text"
      />
      <label
        htmlFor="LoginForm__password"
      />
      <input
        id="LoginForm__password"
        name="password"
        placeholder="Password"
        required={true}
        type="password"
      />
    </fieldset>
    <button
      className="login-button"
      type="submit"
    >
      Login
    </button>
  </form>,
]
`;
